base/threadfuncs.jl

```diff
@@ -0,0 +1,35 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    Threads.foreach(f, channel::Channel; ntasks=Threads.nthreads())
+
+Similar to `foreach(f, channel)`, but iteration over `channel` and calls to
+`f` are split across `ntasks` Tasks spawned by `Threads.@spawn`. This function
+will wait for all internally spawned tasks to complete before returning.
+
+!!! compat "Julia 1.6"
+    This function requires Julia 1.6 or later.
+"""
+function Threads.foreach(f, channel::Channel; ntasks=Threads.nthreads())
```

Comment:
```suggestion
function Threads.foreach(f, channel::Channel; ntasks=Threads.nthreads(),
                         schedule=:default)
    apply = schedule === :static ? (f, x) -> f(x) : _waitspawn
```

(should be `scheduler`?)

Reply:
I prefer `schedule`. 

Also how about `:fair` for the default `schedule` value vs. `:default`?

Reply:
`:fair` sounds good to me.